src/commands/test.rs

```diff
@@ -0,0 +1,325 @@
+//! Implementation of the `test` subcommand.
+
+use std::fs::read;
+use std::path::PathBuf;
+
+use anyhow::Context;
+use anyhow::anyhow;
+use clap::Parser;
+use indexmap::IndexMap;
+use itertools::Itertools;
+use itertools::enumerate;
+use serde_yaml_ng::Value;
+use tracing::info;
+use tracing::trace;
+use tracing::warn;
+
+use crate::analysis::Analysis;
+use crate::analysis::Source;
+use crate::commands::CommandError;
+use crate::commands::CommandResult;
+use crate::test::InputMapping;
+
+/// Arguments for the `test` subcommand.
+#[derive(Parser, Debug)]
+pub struct Args {
+    /// Local path to a WDL document or workspace to unit test.
+    pub source: Option<Source>,
+}
+
+/// A tuple of an input name (`String`) and an input value (as a
+/// [`serde_yaml_ng::Value`] which has not been converted into a WDL value yet)
+type Input = (String, Value);
+/// A map of input keys to values which correspond to a single "run" or
+/// execution for Sprocket to test with. Should be a complete set of required
+/// inputs (potentially with values for optional inputs).
+///
+/// e.g.
+/// {
+///   "bams": ["$FIXTURES/test1.bam", "$FIXTURES/test2.bam"],
+///   "prefix": "test.merged",
+/// }
+type Run = IndexMap<String, Value>;
```

Comment:
Question for here and other places we use mapping data structures: do we actually needs maps or can we store `Vec` in these places?

That is to say: do we expect to lookup entries by key name in these maps?

I think they're generally just iterated over to perform the cartesian product, which itself can produce a set of (name, value) pairs that will be used to construct an `Inputs` to run the workflow or task.